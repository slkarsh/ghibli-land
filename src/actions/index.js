export const setMovies = movies => {
  return ({
    type: 'SET_MOVIES',
    movies
  })
}

export const setPeople = people => {
  return ({
    type: 'SET_PEOPLE',
    people
  })
}

export const setPlaces = places => {
  return ({
    type: 'SET_PLACES',
    places
  })
}

export const setVehicles = vehicles => {
  return ({
    type: 'SET_VEHICLES',
    vehicles
  })
}

export const addCharacter = character => {
  return ({
    type: 'ADD_CHARACTER',
    character
  })
}

export const removeCharacter = character => {
  return ({
    type: 'REMOVE_CHARACTER',
    character
  })
}

export const addLocation = location => {
  return ({
    type: 'ADD_LOCATION',
    location
  })
}

export const removeLocation = location => {
  return ({
    type: 'REMOVE_LOCATION',
    location
  })
}

export const addVehicle = vehicle => {
  return ({
    type: 'ADD_VEHICLE',
    vehicle
  })
}

export const removeVehicle = vehicle => {
  return ({
    type: 'REMOVE_VEHICLE',
    vehicle
  })
}

export const toggleAddCharacter = character => {
  return ({
    type: 'TOGGLE_ADD_CHARACTER',
    character
  })
}