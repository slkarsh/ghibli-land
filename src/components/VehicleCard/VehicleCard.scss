img {
  width: 100%;
}

.vehicle-card {
  padding: 20px;
  border-radius: 6px 6px;
  background: rgba(255, 255, 255, 0.47);
  border: 3px solid #e91eb526;
}