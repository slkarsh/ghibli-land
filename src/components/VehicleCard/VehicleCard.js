import React from 'react'
import './VehicleCard.scss'
import { toggleAddVehicle } from '../../actions'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

const images = {
  Airship: 'https://steamuserimages-a.akamaihd.net/ugc/800921761784638066/E000FAC38E199977DA4A399F9DCEAE53C4134D14/',
  Airplane: 'http://4.bp.blogspot.com/-9LYF-rZrRkE/TbQPOOPWumI/AAAAAAAAAAM/gHcCg6K0KxA/s1600/porco_rosso_movie_image_01.jpg',
  Boat: 'https://vignette.wikia.nocookie.net/studio-ghibli/images/b/b7/Sosuke_in_boat.png/revision/latest?cb=20181024030411'

}

export const VehicleCard = ({ name, description, vehicle_class, toggleAddVehicle, checkVehicles }) => {
  const buttonHandler = checkVehicles(name) ? 'Remove From Your Plot' : 'Add To Your Plot'
  return (
    <div className='vehicle-card'>
      <img src={images[vehicle_class]} />
      <h3>Name: {name}</h3>
      <h4>Vehicle Class: {vehicle_class}</h4>
      <p className='vehicle-description'>{description}</p>
      <button className='vehicle-button' onClick={() => toggleAddVehicle({ name, description, vehicle_class})}>{buttonHandler}</button>
    </div>
  )
}

const mapStateToProps = state => ({
  diyMovie: state.diyMovie
})

const mapDispatchToProps = dispatch => ({
  toggleAddVehicle: vehicle => dispatch( toggleAddVehicle(vehicle) )
})

export default connect(mapStateToProps, mapDispatchToProps)(VehicleCard)

VehicleCard.propTypes = {
  diyMovie: PropTypes.object,
  toggleAddVehicle: PropTypes.func,
  name: PropTypes.string,
  description: PropTypes.string,
  vehicle_class: PropTypes.string,
  checkVehicles: PropTypes.func
}