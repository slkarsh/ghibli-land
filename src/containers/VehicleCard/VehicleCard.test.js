import { VehicleCard, mapStateToProps, mapDispatchToProps } from './VehicleCard'
import React from 'react'
import { shallow } from 'enzyme'
import { toggleAddVehicle } from '../../actions'

describe('VehicleCard', () => {
  let wrapper;
  const mockCheckVehicles = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<VehicleCard
      name='mazda'
      description='a small car'
      vehicle_class='car'
      checkVehicles={mockCheckVehicles}
       />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('mapStateToProps', () => {
    it('should return an object with an array of characters, locations, and vehicles', () => {
      const mockState = {
        diyMovie: {
          characters: [
            {
              name: 'Oliver',
              age: '1.5',
              species: 'dog'
            }
          ],
          locations: [
            {
              name: 'earth',
              terrain: 'varies'
            }
          ],
          vehicles: [
            {
              type: 'car',
              name: 'mazda',
            }
          ]
        }
      }

      const expectedState = {
        diyMovie: {
          characters: [
            {
              name: 'Oliver',
              age: '1.5',
              species: 'dog'
            }
          ],
          locations: [
            {
              name: 'earth',
              terrain: 'varies'
            }
          ],
          vehicles: [
            {
              type: 'car',
              name: 'mazda',
            }
          ]
        }
      }
  
      const mappedProps = mapStateToProps(mockState)
  
      expect(mappedProps).toEqual(expectedState)
    })
  })

  describe('mapDispatchToProps', () => {
    it('calls dispatch with a vehicle object when toggleAddVehicle is called', () => {
      const mockDispatch = jest.fn()

      const actionToDispatch = toggleAddVehicle({name: 'mazda', description: 'a small car', vehicle_class: 'car'})

      const mappedProps = mapDispatchToProps(mockDispatch)

      mappedProps.toggleAddVehicle({name: 'mazda', description: 'a small car', vehicle_class: 'car'})

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})