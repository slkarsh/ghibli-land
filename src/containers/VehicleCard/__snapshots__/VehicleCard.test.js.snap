// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleCard should match the snapshot 1`] = `
<div
  className="vehicle-card"
>
  <img />
  <h3>
    Name: 
    mazda
  </h3>
  <h4>
    Vehicle Class: 
    car
  </h4>
  <p
    className="vehicle-description"
  >
    a small car
  </p>
  <button
    className="vehicle-button"
    onClick={[Function]}
  >
    Add To Your Plot
  </button>
</div>
`;
